src/reporting/Core_json_output.ml

```diff
@@ -223,23 +223,23 @@ let unsafe_match_to_match
     Metavar_replacement.interpolate_metavars x.rule_id.message
       (Metavar_replacement.of_bindings x.env)
   in
-  (* We need to do this, because in Terraform, we may end up with a `file` which
-     does not correspond to the actual location of the tokens. This `file` is
-     erroneous, and should be replaced by the location of the code of the match,
-     if possible. Not if it's fake, though.
-     In other languages, this should hopefully not happen.
-  *)
-  let file =
-    if
-      (!!(x.file) <> min_loc.pos.file || !!(x.file) <> max_loc.pos.file)
-      && min_loc.pos.file <> "FAKE TOKEN LOCATION"
-    then min_loc.pos.file
-    else !!(x.file)
-  in
   {
     check_id = x.rule_id.id;
     (* inherited location *)
-    path = Fpath.v file;
+    path =
+      (match x.source with
+      (* We need to do this, because in Terraform, we may end up with a `file` which
```

Comment:
Would be useful to explain why and in what way we create a fake/erroneous file

Reply:
Yeah, as above this is just moved. I don't have any additional context beyond this comment cc: @brandonspark who wrote this I think.

Reply:
This is because in Terraform, all the files within a particular directory are actually just one big file, semantically. We process it as such, putting all the original file's contents into a fake file, but this causes fake tokens to appear.

The comment could be better.